File: tests/eval_c/273.c
#include <assert.h>
#include <string.h>

void f273(const char* s) {
    char c = s[strlen(s) - 1];
    if (c == 'b') {
        assert(strlen(s) == 2);
        assert(s[0] == 'a');
    } else {
        assert(c == 'a');
        assert(strlen(s) == 1);
    }
}

File: tests/oopsla25/273.py
#!/usr/bin/env python3

import sys

def f273(s: str):
  c = s[len(s)-1]
  if c == "b":
    assert len(s) == 2
    assert s[0] == "a"
  else:
    assert c == "a"
    assert len(s) == 1

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f273(my_string)
